lib/sisimai/bite/email/ezweb.rb

```diff
@@ -0,0 +1,260 @@
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['from']    =~ Re0[:from]
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['received'].find { |a| a =~ Re0[:received] }
+        if mhead['message-id']
+          match += 1 if mhead['message-id'] =~ Re0[:'message-id']
+        end
+        return nil if match < 2
+
+        require 'sisimai/string'
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        rxboundary = %r/\A__SISIMAI_PSEUDO_BOUNDARY__\z/
+        v = nil
+
+        if mhead['content-type']
+          # Get the boundary string and set regular expression for matching with
+          # the boundary string.
+          require 'sisimai/mime'
+          b0 = Sisimai::MIME.boundary(mhead['content-type'], 1)
+          if b0.size > 0
+            # Convert to regular expression
+            rxboundary = Regexp.new('\A' + Regexp.escape(b0) + '\z')
+          end
+        end
+        rxmessages = []
+        ReFailure.each_key { |a| rxmessages.concat(ReFailure[a]) }
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822] || e =~ rxboundary
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # The user(s) account is disabled.
+            #
+            # <***@ezweb.ne.jp>: 550 user unknown (in reply to RCPT TO command)
+            #
+            #  -- OR --
+            # Each of the following recipients was rejected by a remote
+            # mail server.
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:deliverystatus]).zero?` instead of `readcursor & Indicators[:deliverystatus] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->